WordPress/gutenberg
# Implement welcome guide modal

Closes #16315.

Implements the welcome guide modal described in https://github.com/WordPress/gutenberg/issues/16315#issuecomment-523174068.

![guide](https://user-images.githubusercontent.com/612155/69555384-6366f880-0fa3-11ea-9f58-a65f5817e936.gif)

## What's changed 

- There's a new `Guide` component in `@wordpress/nux` which implements the guide modal in a reusable way. This is marked experimental so that we can potentially move this to `@wordpress/components` in a follow-up PR.
- `WelcomeGuide` in `@wordpress/edit-post` will use `Guide` to display a welcome guide for the block editor if the user has not previously disabled tips. This replaces all of our instances of the `DotTip` component. We can potentially move away from using `hasDisabledTips` in a follow-up PR.
- There's a new _Welcome Guide_ button in the More menu which allows the user to open the guide again. This replaces the _Enable Tips_ setting in the Options modal.

## How to test

Unit and E2E tests are included. To manually test the upgrade process, follow these steps:

1. On `master`, open the block editor and ensure that you have dismissed tips.
1. Check out this branch and open the block editor. The welcome guide should not appear.
1. [Rest your browser's `localStorage`](https://developers.google.com/web/tools/chrome-devtools/storage/localstorage) to simulate being a new user.
1. Refresh the page. The guide should appear.
1. Dismiss or complete the guide.
1. Refresh the page. The guide should no longer appear.
